docs/feature_midi.md

```diff
@@ -0,0 +1,254 @@
+# MIDI
+
+## Usage
+
+First, enable MIDI by adding the following to your `rules.mk`:
+
+```makefile
+MIDI_ENABLE = yes
+```
+
+There are two MIDI systems in QMK: basic and advanced. With basic MIDI you will only be able to send Note On and Note Off messages using the note keycodes, meaning that keycodes like `MI_OCTU` and `MI_OCTD` will not work. Advanced MIDI allows you to do things like octave shifts, channel changes, velocity changes, modulation, and more.
+
+### Basic MIDI
+
+To enable basic MIDI, add the following to your `config.h`:
+
+```c
+#define MIDI_BASIC
+```
+
+### Advanced MIDI
+
+To enable advanced MIDI, add the following to your `config.h`:
+
+```c
+#define MIDI_ADVANCED
+```
+
+#### Sending MIDI Control Codes (MIDI CC)
+
+If you're aiming to emulate the features of something like a Launchpad or other MIDI controller you'll need to access the internal MIDI device directly.
+
+When sending MIDI CC you don't get the advantages of a preimplemented keycode and you will need to implement custom keycodes if you want to use them in your keymap directly using the function `process_record_user`. It is the same process as implementing custom keycodes for macros.
+
+For an overview of that process look at: [Macros](feature_macros.md)
```

Comment:
```suggestion
Because there are so many possible CC messages, not all of them are implemented as keycodes. Additionally, you might need to provide more than just two values that you would get from a keycode (pressed and released) - for example, the analog values from a fader or a potentiometer. So, you will need to implement [custom keycodes](feature_macros.md) if you want to use them in your keymap directly using `process_record_user()`.
```